deployment/common/changeset/set_config_mcms_test.go

```diff
@@ -0,0 +1,322 @@
+package changeset_test
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	chain_selectors "github.com/smartcontractkit/chain-selectors"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+
+	commonchangeset "github.com/smartcontractkit/chainlink/deployment/common/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+
+	"github.com/smartcontractkit/chainlink/deployment/common/types"
+	"github.com/smartcontractkit/chainlink/deployment/environment/memory"
+)
+
+// setupSetConfigTestEnv deploys all required contracts for the setConfig MCMS contract call.
+func setupSetConfigTestEnv(t *testing.T) deployment.Environment {
+
+	lggr := logger.TestLogger(t)
+	cfg := memory.MemoryEnvironmentConfig{
+		Nodes:  1,
+		Chains: 2,
+	}
+	env := memory.NewMemoryEnvironment(t, lggr, zapcore.DebugLevel, cfg)
+	chainSelector := env.AllChainSelectors()[0]
+
+	config := proposalutils.SingleGroupTimelockConfig(t)
+	// Deploy MCMS and Timelock
+	env, err := commonchangeset.ApplyChangesets(t, env, nil, []commonchangeset.ChangesetApplication{
+		{
+			Changeset: commonchangeset.WrapChangeSet(commonchangeset.DeployLinkToken),
+			Config:    []uint64{chainSelector},
+		},
+		{
+			Changeset: commonchangeset.WrapChangeSet(commonchangeset.DeployMCMSWithTimelock),
+			Config: map[uint64]types.MCMSWithTimelockConfig{
+				chainSelector: config,
+			},
+		},
+	})
+	require.NoError(t, err)
+	return env
+}
+
+// TestSetConfigMCMS tests the SetConfigMCMS changeset by calling SetConfig and checking the config values.
+func TestSetConfigMCMS(t *testing.T) {
```

Comment:
you'll save some boilerplate code if you do it like this https://github.com/smartcontractkit/chainlink/blob/develop/deployment/ccip/changeset/cs_ccip_home_test.go#L22

Reply:
adjusted here, changed the param to be the changesets instead of a boolean https://github.com/smartcontractkit/chainlink/pull/15684/commits/e16dee8d4a1798d860afcd7db960309f5159f0e5